File: components/src/main/java/net/yakclient/graphics/components/Text.java
package net.yakclient.graphics.components;

import net.yakclient.graphics.api.gui.NativeGuiComponent;
import net.yakclient.graphics.api.gui.GuiProperties;
import net.yakclient.graphics.api.render.FontRenderingContext;
import net.yakclient.graphics.api.render.RenderingContext;

public class Text extends NativeGuiComponent {
    @Override
    public RenderingContext[] glRender(GuiProperties properties) {
        final var specs = new TextSpecProps(properties);

        final FontRenderingContext fontContext = new FontRenderingContext(specs.getFont(), specs.getValue(), specs.getColor(), specs.getX(), specs.getY());

        return this.combineContexts(fontContext, this.renderChildren(properties));
    }
}

File: components/src/main/java/net/yakclient/graphics/components/Text.kt
package net.yakclient.graphics.components

import net.yakclient.graphics.api.gui.GuiProperties
import net.yakclient.graphics.api.gui.NativeGuiComponent
import net.yakclient.graphics.api.render.RenderingContext
import net.yakclient.graphics.opengl.render.FontRenderingContext

public class Text : NativeGuiComponent() {
    public override fun renderNatively(props: GuiProperties): List<RenderingContext> {
        val specs = TextSpecProps(props)
        return this.combine(FontRenderingContext(specs.font, specs.value, specs.color, specs.x, specs.y), this.applyChildren(props))
    }
}